python/paddle/amp/auto_cast.py

```diff
@@ -662,10 +716,12 @@ def master_grad_hook():
 
 
 class StateDictHook:
-    def __init__(self, save_dtype):
+    _save_dtype: dtype
+
+    def __init__(self, save_dtype: dtype) -> None:
```

Comment:
```suggestion
    def __init__(self, save_dtype: str) -> None:
```

根据上下文

``` python
    if save_dtype is not None:
        if save_dtype not in ['float16', 'bfloat16', 'float32', 'float64']:
            raise ValueError(
                f"save_dtype can only be float16 float32 or float64, but your input save_dtype is {save_dtype}."
            )
        for idx in range(len(models)):
            for layer in models[idx].sublayers(include_self=True):
                layer.register_state_dict_hook(StateDictHook(save_dtype))

```

用 str 吧 ～